include/led.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Driver specific representation of a LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable time period (in ms) supported by the device */
```

Comment:
Thanks for adding units, but I'm not sure what this means. Minimum acceptable time period for what? Please see more detailed question about duty cycle and period elsewhere.

Reply:
I think we can use `delay` here than using `period`. This will make the LED API to look distinct from the PWM API. Please see more below.